### PR TITLE
Build a Jupyter book

### LINKED ISSUE #13: Build a Jupyter book
Similar to https://github.com/altair-viz/altair-tutorial. It would make it easier to read the documents and even download them as PDF.

### PR DESCRIPTION
**Pull request by joelostblom**
@domoritz Here is the Jupyter Book PR. I moved the notebooks to the `doc` folder and added CI to rebuild the book upon changes to this folder. You can have a [peak at the rendered book from my branch here](https://joelostblom.github.io/visualization-curriculum/intro.html). I used the logo from your github organization for the book and a slightly modified version of the readme for the landing page (but please edit if this is not appropriate).

close #13

### CODE REVIEW

**Comment by domoritz** [comment]
Fantastic. Thank you. 

Since you moved the files, does that mean the links in the readme don't work anymore? If so, can we load notebooks from the root?

File: doc/_config.yml [hunk 1]
```diff
@@ -0,0 +1,32 @@
+# Book settings
+# Learn more at https://jupyterbook.org/customize/config.html
+
+title: Visualization Curriculum
+author: UW Data
```
Comment (domoritz) [inline doc/_config.yml:5]:
```suggestion
author: Jeffrey Heer, Dominik Moritz, Jake VanderPlas, and Brock Craft
```

File: doc/_config.yml [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+logo: idl-logo.png
```
Comment (domoritz) [inline doc/_config.yml:6]: What do you think about a more generic logo. 

Maybe start with this image from the notebooks

image


[Open the Chart in the Vega Editor](https://vega.github.io/editor/#/url/vega-lite/N4Igxg9gdgZglgcxALlANzgUwO4tAZwBcAnCAa0xRBIEMp8AHG4zKQkAXwBoQaAPOPjwgExOABMAwhAA2EYlVr0mLNp24hxNQjWEBXYjKoALQoQb5kAeithxUAHQArfOMwy4aYg6iZCVqAYAWys0TAQaAFotHXw-fAABAEYrGJorIIgMTHxnfGhOHiDmMioAI2YQHlZIcTgoJFQQPmFCAE8GSmQQAEc9OkI4HUGwqpAy+uFivgn6FAAmAAYNeHdxKgBJAFkAEQAhAH0AJW16pB5+QRQoPRkZDTbWjq7e-rYh09GLhFFw7RfIHo1BcBEJkDc7hwNMZMIhTCgkgBWRY8AAk+DAMOKJjMFmsoT+DgQQ2MejKDjgECsGKx6TCEUiHkImFCABYHAAOBxJPIFDhAA)

Comment (joelostblom) [reply doc/_config.yml:6]: Sounds good! This is how it looks with what you posted and a few variants, let me know which one you prefer:

1.
image

2.
image

3.
image

4.
image

5.
image

**Comment by joelostblom** [comment]
> Since you moved the files, does that mean the links in the readme don't work anymore? If so, can we load notebooks from the root?

Yes, sorry I forgot to update the links. It does not seem like it is possible to reference notebooks outside the root without copying them: https://github.com/executablebooks/jupyter-book/issues/1072. For now, I updated the links in the readme instead.

File: doc/_config.yml (hunk 2 above)
Comment (domoritz) [reply doc/_config.yml:6]: I like 4. with the nod to Altair.

**Comment by domoritz** [comment]
I guess we have to decide whether it's acceptable to change the links. Maybe we could provide symlinks in the root to avoid breaking peoples links.

What do you think @jheer? Do you rely on the current links to the root anywhere?

**Comment by joelostblom** [comment]
Another option is to build the book from the root. That might work better if breaking links is a concern. There are only 4 additional files that would be added to the root (toc, config, intro, logo), so I don't think it would be too confusing.

File: doc/_config.yml (hunk 2 above)
Comment (joelostblom) [reply doc/_config.yml]: Done! I added the VegaEditor URL as a comment in the config file in case anyone wants to modify it in the future.

**Comment by domoritz** [comment]
Let's build from the root.

File: .github/workflows/build-book.yml [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+
+      - name: Set up Python
+        uses: actions/setup-python@v2
+        with:
+          python-version: '3.x'
+
+      - name: Install dependencies
+        run: |
+          python -m pip install --upgrade pip
+          python -m pip install jupyter-book altair vega_datasets
```
Comment (domoritz) [inline .github/workflows/build-book.yml:34]: At some point, we should provide a requirements.txt file with versions. But not needed in this pull request.

File: _config.yml [hunk 4]
```diff
@@ -0,0 +1,32 @@
+# Book settings
+# Learn more at https://jupyterbook.org/customize/config.html
+
+title: Visualization Curriculum
+author: Jeffrey Heer, Dominik Moritz, Jake VanderPlas, and Brock Craft
+logo: altair-hist.png
+# Logo created via https://vega.github.io/editor/#/url/vega-lite/N4Igxg9gdgZglgcxALlANzgUwO4tAZwBcAnCAa0xRBIEMp8AHG4zKQkAXwBoQaAPOPjwgExOABMAwhAA2EYlVr0mLNp24hxNQjWEBXYjKoALQoQb5kAeithxUAHQArfOMwy4aYg6iZCVqAYAWys0TAQaAFotHXw-fAABAEYrGJorIIgMTHxnfGhOHiDmMioAI2YQHlZIcTgoJFRwWXlheHdxKgBJAFkAEQAhAH0AJW16pB4y+uFivmn6FCSABg0ZcNZO5Cg9GRkefDAadeFiOgRKZABtEABiGDKwZYAmTp57x5e3u+WAZhoAJwwAAsVR+-yBoPewJgYF+MAA7GDbjC4YjkWVfpgAKywjFY3FgEAAXR4LDCxDiKBIekwHA0fGEhAAngxLiAAI56OiEOA6XlhMELWb8BZCZDPVY8doyLYgXqDUbjBpg-iCFA7PYHI4nJrkzCUy40ukaZlM1nsrk8vnjQU8GgIUThbTsyB6NT2gTizUyek8YyYRCmJbY5Y8AAkhwDxRMZgs1lCzocCD5xj0ZQccAgVijmGKiYikQ8hEwoWBDgAHA4knkChwgA
```
Comment (domoritz) [inline _config.yml:7]: Nice

**Comment by domoritz** [comment]
Thank you!

**Comment by joelostblom** [comment]
Thanks for merging! I think the only manual thing that is needed after merging is to change the GH settings to point GitHub pages to the gh-pages branch created in the CI. I guess also adding the link to the build book in the repo about box and and maybe also in the repo readme.

**Comment by domoritz** [comment]
On it. I am waiting for GitHub to deploy https://uwdata.github.io/visualization-curriculum/. I think we should also add a link to the readme.

**Pull request merged**